lib/quandl/api_config.rb

```diff
@@ -4,29 +4,36 @@ class ApiConfig
     API_BASE_THREAD_KEY = 'quandl_api_base'
     API_VERSION_THREAD_KEY = 'quandl_api_version_key'
 
+    @api_key = nil
+    @api_base = nil
+    @api_version = nil
+
     class << self
       def api_key=(api_key)
+        @api_key = api_key if @api_key.nil?
```

Comment:
we probably want to maintain as much backwards compatibility as possible now that this is in the open. being able to reset simply by assigning a new value is most idea (i.e., the current versions behaviour)